src/analyses/does_remove_const.cpp

```diff
@@ -122,15 +124,14 @@ bool does_remove_constt::does_type_preserve_const_correctness(
 {
   while(target_type->id()==ID_pointer)
   {
-    bool direct_subtypes_at_least_as_const=
-      is_type_at_least_as_const_as(
-        target_type->subtype(), source_type->subtype());
+    bool direct_subtypes_at_least_as_const = is_type_at_least_as_const_as(
+      to_pointer_type(*target_type).base_type(), source_type->subtype());
     // We have a pointer to something, but the thing it is pointing to can't be
     // modified normally, but can through this pointer
     if(!direct_subtypes_at_least_as_const)
       return false;
     // Check the subtypes if they are pointers
-    target_type=&target_type->subtype();
+    target_type = &to_pointer_type(*target_type).base_type();
```

Comment:
May I suggest a temporary to avoid the repeated `to_pointer_type(*target_type)` in this code branch?

Reply:
done